src/coreComponents/physicsSolvers/SolverStatistics.hpp

```diff
@@ -22,176 +22,377 @@
+  integer m_numSuccessfulLinearIterations = 0;
+
+  /// Cumulative number of discarded configuration iterations
+  integer m_numDiscardedConfigIterations  = 0;
+
+  /// Cumulative number of discarded nonlinear iterations
+  integer m_numDiscardedNonlinearIterations = 0;
+
+  /// Cumulative number of discarded linear iterations
+  integer m_numDiscardedLinearIterations = 0;
+
+  /// Linear solver setup
+  real64 m_setupTime = 0.0;
+
+  /// Linear solver solve
+  real64 m_solveTime = 0.0;
+
   /**
-   * @brief Constructor for SolverStatistics Objects.
-   * @param[in] name the name of this instantiation of SolverStatistics in the repository
-   * @param[in] parent the parent group of this instantiation of SolverStatistics
+   * @brief Struct to serve as a container for variable strings and keys.
+   * @struct viewKeyStruct
    */
-  SolverStatistics( string const & name,
-                    dataRepository::Group * const parent );
+  struct viewKeyStruct
+  {
+    /// String key for the number of time steps
+    static constexpr char const * numTimeStepsString() { return "numTimeSteps"; }
+    /// String key for the number of time step cuts
+    static constexpr char const * numTimeStepCutsString() { return "numTimeStepCuts"; }
+
+    /// String key for the successful number of configuration iterations
+    static constexpr char const * numSuccessfulConfigIterationsString() { return "numSuccessfulConfigIterations"; }
+    /// String key for the successful number of nonlinear iterations
+    static constexpr char const * numSuccessfulNonlinearIterationsString() { return "numSuccessfulNonlinearIterations"; }
+    /// String key for the successful number of linear iterations
+    static constexpr char const * numSuccessfulLinearIterationsString() { return "numSuccessfulLinearIterations"; }
+
+    /// String key for the discarded number of configuration iterations
+    static constexpr char const * numDiscardedConfigIterationsString() { return "numDiscardedConfigIterations"; }
+    /// String key for the discarded number of nonlinear iterations
+    static constexpr char const * numDiscardedNonlinearIterationsString() { return "numDiscardedNonlinearIterations"; }
+    /// String key for the discarded number of linear iterations
+    static constexpr char const * numDiscardedLinearIterationsString() { return "numDiscardedLinearIterations"; }
+  };
+
+  /**
+   * @brief Set the csv state output
+   * @param state The csv state
+   */
+  void setCSVOutput( bool state )
+  { m_csvOutput = state; }
+
+  /**
+   * @brief Set the log state output
+   * @param state The log state
+   */
+  void setLogOutput( bool state )
+  { m_logOutput = state; }
+
 
   /**
    * @brief Initialize the counters used for an individual time step
    */
-  void initializeTimeStepStatistics();
+  void resetCurrentTimeStepStatistics();
+
+  /**
+   * @brief Tell the solverStatistics that we are doing a configuration iteration
```

Comment:
```suggestion
   * @brief Tell the solverStatistics that we have done a configuration iteration
```
so it is really specifying how we should use this class?

Reply:
same remark for all your explaining comments in `SolverStatistics.cpp`, they can give information to class users